crates/api_crud/src/post/update.rs

```diff
@@ -41,6 +41,10 @@ impl PerformCrud for EditPost {
     let local_user_view =
       get_local_user_view_from_jwt(&data.auth, context.pool(), context.secret()).await?;
 
+    let data_url = data.url.as_ref();
+    let url = Some(data_url.map(clean_url_params).map(Into::into)); // TODO no good way to handle a "clear"
```

Comment:
Whats the todo about?

Reply:
So the problem is this: how do I clear a URL field?

With the simple optional `String` fields like body, you can send an empty string `""`, and use that to tell diesel to clear that field, by doing a `Some(None)`. IE 

- Don't update the body: `None`
- Clear the body: `Some(None)`

But since you've deserialized the JSON string into Option<URL> with serde, there's no way to send an empty string. So it has no way to differentiate between these two actions:

- Don't update the URL
- Clear the URL

Reply:
Right, theres an issue for that so i would just put the issue link in the code. 

I also cant really think about a solution, other than an explicit field like `clear_url: bool`.

Reply:
What's the issue? I can't find.

What I'd recommend, is doing what we do for avatar and banner, having that field as a String which then can use `diesel_option_overwrite_to_url`

Reply:
https://github.com/LemmyNet/lemmy/issues/2287